lib/ajax-datatables-rails/orm/active_record.rb

```diff
@@ -38,31 +38,28 @@ def composite_search(records)
         records
       end
 
-      def build_conditions_for(query)
+      def build_conditions_for(query, regex)
         search_for = query.split(' ')
         criteria = search_for.inject([]) do |criteria, atom|
-          criteria << searchable_columns.map { |col| search_condition(col, atom) }
+          criteria << searchable_columns.map { |col| search_condition(col, atom, regex == 'true') }
             .reduce(:or)
         end.reduce(:and)
         criteria
       end
 
       def aggregate_query
-        conditions = searchable_columns.each_with_index.map do |column, index|
+        conditions = view_columns.each_with_index.map do |column, index|
```

Comment:
Also, by using the whole `view_columns` array here... we'll be ignoring the columns description coming from the params hash... again, from your example where columns 0,1,2,4 are searchable and 3 is not... by using `view_columns` we'll be using _all_ columns (0,1,2,3,4) to build our `where` clauses... then, what would be point to mark 3 as non-searchable?